src/ops.rs

```diff
@@ -915,6 +938,232 @@ mod kmac_tests {
     }
 }
 
+#[cfg(test)]
+mod decryption_test {
+    // Ensure to test if there are if & else cases: write two tests for each if and else case
+    use crate::{
+        sha3::aux_functions::byte_utils::get_random_bytes, 
+        KeyEncryptable, 
+        KeyPair, 
+        Message, 
+        SecParam::{D224, D256, D384, D512}, 
+        SpongeEncryptable
+    };
+    #[test]
+    ///
+    /// Testing if the failed decryption preserves the same message
+    /// if password1 not_equal password2,
+    /// then the original encrypted message should be the same.
+    fn test_sha3_decrypt_handling_bad_input() {
+        let mut new_msg = Message::new(get_random_bytes(523)); // mutable
+
+        // create a copy of message, and pass the copy of message into the encryption & decryption
+        let pw1 = get_random_bytes(64);
+        let pw2 = get_random_bytes(64);
+        // encrypted_message with password1
+        new_msg.sha3_encrypt(&pw1, &D512);
+
+        // cloned after encryption for testing purpose only
+        let msg2 = new_msg.msg.clone(); 
+
+        // decrypt message with password2
+        new_msg.sha3_decrypt(&pw2);
+        // after decryption, message is still preserved without being affected
```

Comment:
this is tested elsewhere in this library and isnt needed here. try to keep this test case solely centered on testing failed decryption